### ISSUE TITLE
Investigate replacing go-wasmer with wasmtime

### ISSUE THREAD

**Issue by amerameen** [ChainSafe/gossamer#1102]
## Expected Behavior

- Currently we are blocked by a bug in go-wasmer, so we should investigate the feasiblity of other options like wasmtime

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I am planning to submit a pull request to fix this issue myself

**Comment by syrusakbary** [comment]
Note: we just finished upgrading our Python integration, and the next one on line is Go!
Stay tuned for updates, a very important upgrade will be coming soon!

**Comment by amerameen** [comment]
Thanks for letting us know @syrusakbary! What is the approx. timeframe can we expect the Go upgrade in?

**Comment by noot** [comment]
hey @syrusakbary thanks for the response. just to clarify, the reason we are considering switching is since wasmer is a crucial part of the project, the bug we discovered back in June has been blocking us for quite a while. it would have been nice to have helped speed it up any way possible. for now, we will continue investigating wasmtime so we can eventually benchmark the two interpreters. looking forward to testing the new wasmer release :)
